### PR TITLE
[SPARK-21878] [SQL] [TEST] Create SQLMetricsTestUtils

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
Creates `SQLMetricsTestUtils` for the utility functions of both Hive-specific and the other SQLMetrics test cases. 

Also, move two SQLMetrics test cases from sql/hive to sql/core. 

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @cloud-fan

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLMetricsSuite.scala [hunk 1]
```diff
@@ -17,112 +17,10 @@
...
-      }
-      assert(Utils.recursiveList(dir).count(_.getName.startsWith("part-")) == 40)
-    }
-  }
-
-  test("writing data out metrics: parquet") {
-    testMetricsNonDynamicPartition("parquet", "t1")
-  }
-
-  test("writing data out metrics with dynamic partition: parquet") {
```
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLMetricsSuite.scala]: The above two test cases are moved to `sql/execution/metric/SQLMetricsSuite.scala`

**Comment by gatorsmile** [comment]
retest this please

**Comment by cloud-fan** [comment]
LGTM

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Pull request closed without merging**